kedro/config/omegaconf_config.py

```diff
@@ -316,31 +318,28 @@ def _register_globals_resolver(self):
         """Register the globals resolver"""
         OmegaConf.register_new_resolver(
             "globals",
-            lambda variable, default_value=None: self._get_globals_value(
-                variable, default_value
-            ),
+            self._get_globals_value,
             replace=True,
         )
 
-    def _get_globals_value(self, variable, default_value):
+    def _get_globals_value(self, variable, default_value=_NO_VALUE):
         """Return the globals values to the resolver"""
         if variable.startswith("_"):
             raise InterpolationResolutionError(
                 "Keys starting with '_' are not supported for globals."
             )
-        keys = variable.split(".")
-        value = self["globals"]
-        for k in keys:
-            value = value.get(k)
-            if not value:
-                if default_value:
-                    _config_logger.debug(
-                        f"Using the default value for the global variable {variable}."
-                    )
-                    return default_value
-                msg = f"Globals key '{variable}' not found and no default value provided. "
-                raise InterpolationResolutionError(msg)
-        return value
+        global_omegaconf = OmegaConf.create(self["globals"])
+        interpolated_value = OmegaConf.select(
+            global_omegaconf, variable, default=default_value
+        )
+        if interpolated_value != _NO_VALUE:
+            return interpolated_value
+        else:
+            raise InterpolationResolutionError(
+                "Default value is not defined for {$globals: {keys}}.".replace(
```

Comment:
Yes. Currently there are 3 paths it can take:
1. exists in `globals.yml` (take the global value as is)
2. Not define in `globals.yml`, but default provided
3. Not define in `globals.yml` and not default provided - raise `InterpolationResolutionError`

Reply:
Done